app/scripts/modules/core/src/pipeline/details/SingleExecutionDetails.tsx

```diff
@@ -38,108 +30,155 @@ export interface ISingleExecutionRouterStateChange extends IStateChange {
-          this.executionScheduler.unsubscribe();
-          this.executionLoader.unsubscribe();
-        }
-        this.setState({ execution });
-
-        app.pipelineConfigs.activate();
-        app.pipelineConfigs.ready().then(() => {
-          const pipelineConfig = app.pipelineConfigs.data.find((p: IPipeline) => p.id === execution.pipelineConfigId);
-          this.setState({ pipelineConfig });
-        });
-      },
-      () => {
-        this.setState({ execution: null, stateNotFound: true });
-      },
-    );
-  }
+  const getAncestry = (execution: IExecution): Promise<IExecution[]> => {
+    const youngest = ancestryRef.current[ancestryRef.current.length - 1];
+    // youngest and execution don't match only during navigating between executions
+    const navigating = execution && youngest && youngest.id !== execution.id;
+    const lineage = traverseLineage(execution);
 
-  public componentDidMount(): void {
-    this.stateChangeSuccessSubscription = ReactInjector.stateEvents.stateChangeSuccess.subscribe(
-      (stateChange: ISingleExecutionRouterStateChange) => {
-        if (
-          !stateChange.to.name.includes('pipelineConfig') &&
-          !stateChange.to.name.includes('executions') &&
-          (stateChange.toParams.application !== stateChange.fromParams.application ||
-            stateChange.toParams.executionId !== stateChange.fromParams.executionId)
-        ) {
-          this.getExecution();
+    // used when navigating between executions so clicking between generations is snappy
+    const ancestryCache = ancestryRef.current.reduce((acc, curr) => set(acc, curr.id, curr), {
+      [execution.id]: execution,
+    });
+
+    // used to skip re-fetching ancestors that are no longer active
+    const inactiveCache = ancestryRef.current
+      .filter((ancestor) => !ancestor.isActive)
+      .reduce((acc, curr) => set(acc, curr.id, curr), { [execution.id]: execution });
+
+    const cache = navigating ? ancestryCache : inactiveCache;
+
+    return Promise.all(
+      lineage.map((generation) =>
+        cache[generation] ? Promise.resolve(cache[generation]) : getAndTransformExecution(generation, app),
+      ),
+    ).then((ancestry) => {
+      ancestryRef.current = ancestry;
+      return ancestry;
+    });
+  };
+
+  // responsible for getting execution whenever executionId (route param) changes
+  const { result: execution, status: getExecutionStatus, refresh: refreshExecution } = useLatestPromise(
+    () => getAndTransformExecution(executionId, app),
+    [executionId],
+  );
+
+  // responsible for getting ancestry whenever execution changes or refreshes
+  const { result: ancestry } = useData(() => getAncestry(execution), [], [execution, executionId]);
+
+  // Manages the scheduled refresh until the entire lineage has no active executions
+  const someActive = [execution]
+    .concat(ancestry)
+    .filter((x) => x)
+    .some((x) => x.isActive);
+  useEffect(() => {
```

Comment:
Ah that would be quite nice.

Unfortunately I don't think I can because I would like to conditionally poll based on `someActive` which is computed from `execution` and `ancestry` (which depends on `execution`).